TESTS/netsocket/udp/udpsocket_echotest_burst.cpp

```diff
@@ -100,7 +100,9 @@ void UDPSOCKET_ECHOTEST_BURST()
             if (check_oversized_packets(sent, tx_buffers[x].len)) {
                 TEST_IGNORE_MESSAGE("This device does not handle oversized packets");
             }
-            TEST_ASSERT_EQUAL(tx_buffers[x].len, sent);
+            if (sent != NSAPI_ERROR_NO_MEMORY) {
```

Comment:
I think I'd be even more relaxed on this - I'd let it return any error, or the size sent. 

But this is fine, as it's relaxing it somewhat.

(The code handling oversized packets seems wrong to me, but that's another issue).